core/shared/src/main/scala/zio/clock/package.scala

```diff
@@ -27,18 +27,41 @@ package object clock {
 
   object Clock extends PlatformSpecific with Serializable {
     trait Service extends Serializable {
+
       def currentTime(unit: TimeUnit): UIO[Long]
+
+      // Could be UIO. We keep IO to preserve binary compatibility.
       def currentDateTime: IO[DateTimeException, OffsetDateTime]
-      def instant: UIO[java.time.Instant]                               = currentTime(TimeUnit.MILLISECONDS).map(java.time.Instant.ofEpochMilli(_))
+
+      // The implementation is only here to preserve binary compatibility.
+      def instant: UIO[java.time.Instant] = currentTime(TimeUnit.MILLISECONDS).map(java.time.Instant.ofEpochMilli(_))
+
+      // This could be a UIO. We keep IO to preserve binary compatibility.
+      // The implementation is only here to preserve binary compatibility.
       def localDateTime: IO[DateTimeException, java.time.LocalDateTime] = currentDateTime.map(_.toLocalDateTime())
```

Comment:
Same comment as above.

Reply:
The previous implementation was based on `currentDateTime`, which could a `DateTimeException`. The new implementation directly uses `LocalDateTime.now`, which does not throw according to the docs.